### PR TITLE
Add confirmation prompt for playground uploads

### LINKED ISSUE #267: Prevent accidental postings to the playground
We are getting about one request of takedown from play.golang.org at security@golang.org per week, all due to the Emacs feature. It suggests users commonly use this feature by mistake.

It would be nice if the shortcut could be changed, removed or a confirmation step added.

### PR DESCRIPTION
**Pull request by psanford**
go-play-region and go-play-buffer will now ask for confirmation before
uploading to the playground. This is to help prevent accidental
uploads of private code that should not be posted to the playground.

You can disable the prompt by setting go-confirm-playground-uploads to
nil.

This fixes #267

### CODE REVIEW

File: go-mode.el [hunk 1]
```diff
@@ -211,6 +211,14 @@ mis-identifying them as gb projects."
   :type '(repeat function)
   :group 'go)
 
+(defcustom go-confirm-playground-uploads t
+  "Ask before uploading code to the public go Playground.
```
Comment (dominikh) [inline go-mode.el]: It's `Go`, not `go`. Same typo on line 1292

Comment (psanford) [reply go-mode.el]: Fixed.

File: go-mode.el [hunk 2]
```diff
@@ -1273,27 +1281,36 @@ declaration."
...
-                          ((equal :error (car arg))
-                           (signal 'go-play-error (cdr arg)))
-                          (t
-                           (re-search-forward "\n\n")
-                           (let ((url (format "https://play.golang.org/p/%s"
-                                              (buffer-substring (point) (point-max)))))
-                             (when go-play-browse-function
-                               (funcall go-play-browse-function url)))))))))))
+  (if (and go-confirm-playground-uploads
+             (not (yes-or-no-p "Upload to public Go Playground?")))
```
Comment (muirrn) [inline go-mode.el]: Is indentation right here?

Comment (psanford) [reply go-mode.el]: Fixed.

**Pull request closed without merging**